src/lib/Libtpp/tpp_transport.c

```diff
@@ -1626,9 +1627,9 @@ work(void *v)
 #endif
 	tpp_log_func(LOG_CRIT, NULL, "Thread ready");
 
-	/* store the log and avl tls data so we can free later */
+	/* store the log and index tls data so we can free later */
 	td->tpp_tls->log_data = log_get_tls_data();
-	td->tpp_tls->avl_data = get_avl_tls();
+	td->tpp_tls->idx_data = pbs_idx_get_tls();
```

Comment:
I think we should get rid of this and instead add a piece code with pthread_once() in idx_create() or one such entry function, so the tls will get created for the process once. We don't actually need it inside TPP, and we do not need to free it when exiting from programs